src/test/compile-fail/borrowck/borrowck-issue-14498.rs

```diff
@@ -27,7 +27,7 @@ fn indirect_write_to_imm_box() {
     let y: Box<_> = box &mut x;
     let p = &y;
     ***p = 2; //[ast]~ ERROR cannot assign to data in a `&` reference
-              //[mir]~^ ERROR cannot assign to immutable item `***p`
+              //[mir]~^ ERROR cannot assign through `&`-reference `p`
```

Comment:
Not convinced about this change...

Reply:
yes, due to the usage here `crate trait FindAssignments {`

Reply:
certainly better than it was ("imutable item"), but not I think perfect yet -- and not necessarily better than AST variant, I don't think

Reply:
The AST based error is this:

```
error[E0389]: cannot assign to data in a `&` reference
 --> src/main.rs:8:5
  |
8 |     ***p = 2;
  |     ^^^^^^^^ assignment into an immutable reference
```

which I think has too much jargon, but some promising elements. This also intersects your style guide @estebank. **Personally**, my preference is that the "main" message is kind of generic and formal, like it is here, telling you nothing of the program itself, but that those details are revealed in the "underlines" and in a "conversational" tone. Maybe something like this?

```
error[E0389]: cannot assign to data in a `&` reference
 --> src/main.rs:8:5
  |
8 |     ***p = 2;
  |     ^^^^^^^^ `p` is a `&` reference, so the data it refers to cannot be written
```

Reply:
Is this the common message for all such cases or just for the above case?

Reply:
Is this the common message for all such cases or just for the above case?